docs/DeveloperGuide.md

```diff
@@ -7,23 +7,99 @@
+     e.g., NAME, AMOUNT, EXPIRATION, PRICE.
+   * Then create a method in "GroceryList" class that prints all the groceries in the list.
+
+
+### 2. List the groceries by price in descending order
+   * First, create a field in "Grocery" class that stores the cost of a grocery.
+   * When adding a grocery, prompt the user to enter the cost.
+   * Format the cost into 2 decimal places, remove the dollar sign and store it as a string.
+   * Second, the grocery's cost accordingly before adding it into the list.
+   * When setting the cost, convert the cost from String into Double.
+   * Third, add method in "GroceryList" class to create a copy of the current grocery list, then sort the 
+   new grocery list by price using lambda function. Reverse and print the new list.
+
+
+### 3. Input category for each grocery added
+   * In Grocery class, modified the Grocery constructor to accept the 'category' parameter.
+   * In Parser class executeCommand method, modified the add command to prompt the user for the category of the grocery. Passed the category as a parameter when creating a new Grocery object.
+   * In Ui class, added a new method promptForCategory to prompt the user for the category of the grocery.
+   * In Grocery class, modified the printGrocery method to include the category information in the output string.
+
+
+### 4. Input amount for each grocery added
+   * In Grocery class, modified the Grocery constructor to accept the 'amount' parameter.
+   * In Parser class executeCommand method, modified the add command to prompt the user for the amount of grocery. Passed the amount as a parameter when creating a new Grocery object.
+   * In Ui class, added a new method promptForAmount to prompt the user for the amount of grocery.
+   * In Grocery class, modified the printGrocery method to print different units of measurement for different categories.
+
+
+### 5. Input the location of where each grocery is stored
+   * In Grocery class, modified the Grocery class to include location (String) as an attribute.
+   * In Grocery class, modified the Grocery constructor to accept the 'location' parameter.
+   * In Grocery class, under printGrocery, added locationString to format location.
+   * In Parser class executeCommand method, modified the add command to prompt the user for where the grocery is stored. Passed the location as a parameter when creating a new Grocery object.
+   * In Ui class, added promptForLocation method to take in user input for location of the grocery.
+   * In Ui class, modified the printGrocery method to print the 'location' of the grocery alongside the grocery name.
+   * Alternative considered: Can possibly add location as enumeration however different people might store groceries in different places thus better to set as String so that user is free to input location details however specific they want.
+
+### 6. Edit grocery amount after using a grocery
```

Comment:
Excellent work on UML, however could your show how the user exits the app?
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/97681546/7662a550-aed6-4017-8948-f8c458aafb25)